### ISSUE TITLE
How to return saved results path

### ISSUE THREAD

**Issue by aatrox9029** [ultralytics/ultralytics#1329]
### Search before asking

- [x] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

`results=model.predict(source=data_path,save=True,save_txt=True,show=True)
`
![image](https://user-images.githubusercontent.com/75787807/223774539-3e9e5b16-a4f7-4844-8784-bb57745bb3ab.png)
How to get this path?

**Comment by glenn-jocher** [comment]
@aatrox9029 hi, thanks for asking! We don't have a simple way to handle this currently. 

@AyushExel we should add this to our feature request list.

**Comment by aatrox9029** [comment]
> @aatrox9029 hi, thanks for asking! We don't have a simple way to handle this currently.
> 
> @AyushExel we should add this to our feature request list.

thank you

**Comment by glenn-jocher** [comment]
@aatrox9029 Great news 😃! Your original issue may now be resolved ✅ in PR #3240. Apologies for the long delay. `save_dir` is now returned inside each prediction `Results` object, i.e.:

```python
from ultralytics import YOLO

model=YOLO()
results=model(save=True)
print(results[0])

....
        [ 96,  86,  92],
        [ 98,  88,  94]]], dtype=uint8)
orig_shape: (1080, 810)
path: '/Users/glennjocher/PycharmProjects/ultralytics/ultralytics/assets/bus.jpg'
probs: None
save_dir: '/Users/glennjocher/PycharmProjects/ultralytics/runs/detect/predict11'
speed: {'preprocess': 1.6410350799560547, 'inference': 50.203800201416016, 'postprocess': 0.5319118499755859}
```

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
